modules/assim.sequential/R/debias_step.R

```diff
@@ -0,0 +1,691 @@
+    }
+    inconsistent_sites <- debias_sites_inconsistent_obs(obs.mean, t_idx, name_map = debias_name_map)
+    eligible_sites <- setdiff(complete_sites, inconsistent_sites)
+  } else {
+    eligible_sites <- complete_sites
+  }
+  
+  if (length(eligible_sites) == 0) {
+    return(tibble::tibble(site = character(0), year = integer(0)))
+  }
+  
+  df_year <- covariates_df[
+    covariates_df$year == as.integer(yr) & covariates_df$site %in% eligible_sites,
+    , drop = FALSE
+  ]
+  df_year <- df_year[order(df_year$site), , drop = FALSE]
+  
+  # Annotate drops for diagnostics/logging
+  attr(df_year, "dropped_missing_covariates") <- setdiff(sites_used, complete_sites)
+  if (isTRUE(enforce_consistent_obs)) {
+    attr(df_year, "dropped_inconsistent_obs") <- intersect(sites_used, debias_sites_inconsistent_obs(obs.mean, t_idx))
+  }
+  
+  df_year
+}
+
+#' Expand per-site covariates to **row-per-column** alignment
+#'
+#' Converts the per-site covariate data into a matrix aligned with the **columns of `X`**.
+#' For any column whose site was filtered out, the function inserts a row of `NA`
+#' features to preserve shape and ordering.
+#'
+#' @param covariates_df See `debias_get_covariates_for_date()`.
+#' @param obs_date Date for which to fetch covariates (year is used).
+#' @param site_index Site id per column of `X`.
+#' @param obs.mean Observation structure used if enforcing consistency.
+#' @param t_idx Time index associated with `obs_date`.
+#' @param drop_incomplete_covariates, enforce_consistent_obs See above.
+#'
+#' @return A numeric matrix with **nrow = length(site_index)** and one column per
+#'   covariate feature. Rows align 1:1 with the columns of `X`.
+#'
+#' @rdname debias_helpers
+#' @keywords internal
+debias_cov_by_columns <- function(covariates_df,
+                                  obs_date,
+                                  site_index,
+                                  obs.mean,
+                                  t_idx,
+                                  drop_incomplete_covariates = TRUE,
+                                  enforce_consistent_obs = TRUE) {
+  df_year <- debias_get_covariates_for_date(
+    covariates_df = covariates_df,
+    obs_date      = obs_date,
+    site_index    = site_index,
+    obs.mean      = obs.mean,
+    t_idx         = t_idx,
+    drop_incomplete_covariates = drop_incomplete_covariates,
+    enforce_consistent_obs     = enforce_consistent_obs
+  )
+  
+  if (nrow(df_year) == 0) {
+    # Preserve outer shape; no features to return.
+    return(matrix(numeric(0), nrow = length(site_index), ncol = 0))
+  }
+  
+  feat_cols <- setdiff(names(df_year), c("site", "year"))
+  idx <- match(as.character(site_index), df_year$site)
+  
+  # Filler row of NA to keep alignment when a site is missing
+  na_row <- as.list(rep(NA_real_, length(feat_cols)))
+  names(na_row) <- feat_cols
+  filler <- tibble::as_tibble_row(na_row)
```

Comment:
```suggestion
  filler <- dplyr::as_tibble_row(na_row)
```

Reply:
Turns out dplyr doesn't export as_tibble_row -- would just as_tibble(na_row) work? Looks like the main thing as_tibble_row gives you is guaranteed complaint if the inputs aren’t all exactly length one, but given you just constructed na_row  I think that’s covered in this particular case
```suggestion
  filler <- dplyr::as_tibble(na_row)
```